src/pages/home/home.ts

```diff
@@ -1,133 +1,233 @@
+      })
+      .catch(error => error);
   }
 
   getCurrentPosition() {
-
-    var options = {
+    const options = {
       enableHighAccuracy: true
     };
-
     return new Promise((resolve, reject) => {
-      navigator.geolocation.getCurrentPosition(resolve, reject, options)
+      navigator.geolocation.getCurrentPosition(resolve, reject, options);
+    });
+  }
+
+  getCentralPosition(currentPostion: Coordinates) {
+    const locations = [currentPostion, ...RANDOM_GEOCOORDINATES];
+
+    return new Promise(resolve => {
+      const centralPosition = geolib.getCenterOfBounds(locations);
+      this.dropMarker(
+        "Central Location",
+        "purple",
+        centralPosition.latitude,
+        centralPosition.longitude
+      );
+      return resolve(centralPosition);
     });
+  }
+
+  getWorkfromLocations(centralPosition) {
+    const { latitude, longitude } = centralPosition;
 
+    this.workfromService.getPlaces(latitude, longitude).subscribe(res => {
+      const locations = res.json();
 
+      if (locations.length > 0) {
+        locations.forEach(location => {
+          this.dropMarker(
+            location.title,
+            "red",
+            location.latitude,
+            location.longitude
+          );
+        });
+      } else {
+        alert(
+          `There aren't any Workfrom locations... We might need to search on Yelp then!`
+        );
+      }
+    });
   }
 
-  getWorkfromLocations(position) {
-    this.workfromService.getPlaces(
-      position.coords.latitude,
-      position.coords.longitude
-    ).subscribe(res => {
-
-      var locations = res.json();
-      locations.forEach(location => {
-        this.map
-        .addMarker({
-          title: location.title,
-          icon: 'blue',
-          animation: 'DROP',
-          position: {
-            lat: location.latitude,
-            lng: location.longitude,
-          },
-        })
-        .then(marker => {
-          marker.on(GoogleMapsEvent.MARKER_CLICK).subscribe(() => {
-            alert('clicked');
-          });
+  showCreateSpaceModal() {
+    //On open space modal, subscribe to group uid from server.
+    this.groupSocketService.uid.subscribe(
+      //User's info object that will be sent to server.
+      group_uid => {
+        this.groupSocketService.userInfo  = {
+          socketID : "",
+          groupUID : group_uid,
+          username: this.username,
+          latitude: this.latitude,
+          longitude: this.longitude
+        };
+
+        console.log(group_uid);
+
+        //Join the room specified by the group uid.
+        this.groupSocketService.joinGroup();
+
+        //Create modal.
+        let spaceModal = this.modalCtrl.create(SpacePage, {
+          uid: group_uid
         });
-      });
 
-    });
+        spaceModal.present();
+
+        spaceModal.onDidDismiss(data => {
+          this.groupSocketService.userInfos = [];
+        });
+      },
+      error => console.log(error)
```

Comment:
So I see that when there's an error, we don't open the modal. That is misleading to the user. We should still open it and just display some text on the modal like, "Sorry, we couldn't create your space link right now, please try again later."

Reply:
Modal opens and provides error text on an empty group uid now.